WooCommerce/Classes/GoogleAds/GoogleAdsEligibilityChecker.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Experiments
+import Yosemite
+
+/// Interface for checking if a site is eligible for creating Google ads campaigns from the app.
+///
+protocol GoogleAdsEligibilityChecker {
+    @MainActor
+    func isSiteEligible(siteID: Int64) async -> Bool
+}
+
+final class DefaultGoogleAdsEligibilityChecker: GoogleAdsEligibilityChecker {
+
+    private let stores: StoresManager
+    private let featureFlagService: FeatureFlagService
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.stores = stores
+        self.featureFlagService = featureFlagService
+    }
+
+    func isSiteEligible(siteID: Int64) async -> Bool {
```

Comment:
❓ Should we mark this as `MainActor`? Also, I think marking the protocol method as `MainActor` doesn't really check the `MainActor` conformance here.

Reply:
@selanthiraiyan Actually marking the protocol method with `@MainActor` ensures that all conformance method are on the main actor. I learned this when attempting to do the [migration for Swift 6](https://github.com/woocommerce/woocommerce-ios/pull/13084), so I figured it's better to do it correctly now than to fix it later in the migration process.

Reply:
> Actually marking the protocol method with @MainActor ensures that all conformance method are on the main actor. 

That is great. TIL. Thanks.